Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/social/TwitterRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,216 @@
+                            ""
+                        )
+                    ).transform(CircleTransform(context)).crossFade().into(profileImage)
+                    screenName.text = tweet.tweet.screenName
+                    name.text = String.format(
+                        context.getString(R.string.twitter_handel),
+                        tweet.tweet.name
+                    )
+                    content.text = tweet.tweet.text
+
+                    if (tweet.media.isNullOrEmpty()) {
+                        mediaContainer.visibility = View.GONE
+                    } else {
+                        renderMedia(mediaContainer, tweet.media!!, onMediaClickListener)
+                    }
+                } else {
+                    // Removing '_normal' in profile image url because it's a low resolution image and
+                    // will look blurry. There is no alternative solution for this and twitter
+                    // recommends this. Url after removing `_normal` gives high resolution image.
+                    Glide.with(context).load(
+                        tweet.tweet.profileImageUrl.replace(
+                            "_normal",
+                            ""
+                        )
+                    ).transform(CircleTransform(context)).crossFade().into(quotedProfileImage)
+                    quotedScreenName.text = tweet.tweet.screenName
+                    quotedName.text = String.format(
+                        context.getString(R.string.twitter_handel),
+                        tweet.tweet.name
+                    )
+                    quotedContent.text = tweet.tweet.text
+                    if (tweet.media.isNullOrEmpty()) {
+                        quotedMediaContainer.visibility = View.GONE
+                    } else {
+                        renderMedia(quotedMediaContainer, tweet.media!!, onMediaClickListener)
+                    }
+                    quotedTweetScreenName.text = tweet.tweet.quotedTweet?.screenName
+                    quotedTweetName.text = String.format(
+                        context.getString(R.string.twitter_handel),
+                        tweet.tweet.quotedTweet?.name
+                    )
+                    quotedTweetContent.text = tweet.tweet.quotedTweet?.text
+
+                    if (tweet.quotedMedia.isNullOrEmpty()) {
+                        quotedTweetMediaContainer.visibility = View.GONE
+                    } else {
+                        renderMedia(
+                            quotedTweetMediaContainer, tweet.quotedMedia!!,
+                            onMediaClickListener
+                        )
+                    }
+                }
+            }
+        }
+
+        private fun renderMedia(
+            mediaContainer: ViewGroup,
+            media: List<Media>,
+            onMediaClickListener: OnMediaClickListener
+        ) {
+            mediaContainer.visibility = View.VISIBLE
+            when (media.size) {
```

Comment:
Another case where I think we can break out the when blocks into individual methods for readability, but not something I want to be too nit picky about.

Reply:
Done